builder/src/lib.rs

```diff
@@ -0,0 +1,1317 @@
+#![allow(unused_imports)]
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    signers::{coins_bip39::English, MnemonicBuilder, Signer as _, Wallet},
+    types::{Address, U256},
+};
+use futures::{
+    future::{join_all, Future},
+    stream::{Stream, StreamExt},
+};
+use hotshot::{
+    traits::{
+        election::static_committee::{GeneralStaticCommittee, StaticElectionConfig},
+        implementations::{NetworkingMetricsValue, WebServerNetwork},
+    },
+    types::{SignatureKey, SystemContextHandle},
+    HotShotInitializer, Memberships, Networks, SystemContext,
+};
+use hotshot_orchestrator::{
+    client::{OrchestratorClient, ValidatorArgs},
+    config::NetworkConfig,
+};
+use hotshot_types::{
+    consensus::ConsensusMetricsValue,
+    event::Event,
+    light_client::StateKeyPair,
+    signature_key::{BLSPrivKey, BLSPubKey},
+    traits::election::Membership,
+    traits::metrics::Metrics,
+    HotShotConfig, PeerConfig, ValidatorConfig,
+};
+use std::fmt::Display;
+// Should move `STAKE_TABLE_CAPACITY` in the sequencer repo when we have variate stake table support
+use hotshot_stake_table::config::STAKE_TABLE_CAPACITY;
+
+use async_std::{
+    sync::Arc,
+    task::{spawn, JoinHandle},
+};
+use hotshot_state_prover;
+use jf_primitives::{
+    merkle_tree::{namespaced_merkle_tree::NamespacedMerkleTreeScheme, MerkleTreeScheme},
+    signatures::bls_over_bn254::VerKey,
+};
+use sequencer::catchup::mock::MockStateCatchup;
+use sequencer::{
+    catchup::StatePeers,
+    context::{Consensus, SequencerContext},
+    l1_client::L1Client,
+    network,
+    persistence::SequencerPersistence,
+    state::FeeAccount,
+    state::ValidatedState,
+    state_signature::{static_stake_table_commitment, StateSigner},
+    BuilderParams, L1Params, NetworkParams, Node, NodeState, PrivKey, PubKey, SeqTypes,
+};
+use std::{alloc::System, any, fmt::Debug, mem};
+use std::{marker::PhantomData, net::IpAddr};
+use std::{net::Ipv4Addr, thread::Builder};
+use tide_disco::{app, method::ReadState, App, Url};
+use versioned_binary_serialization::version::StaticVersionType;
+type ElectionConfig = StaticElectionConfig;
+
+pub struct BuilderContext<N: network::Type, Ver: StaticVersionType + 'static> {
+    /// The consensus handle
+    pub hotshot_handle: Consensus<N>,
+
+    /// Index of this sequencer node
+    pub node_index: u64,
+
+    /// Context for generating state signatures.
+    state_signer: Arc<StateSigner<Ver>>,
+
+    /// An orchestrator to wait for before starting consensus.
+    pub wait_for_orchestrator: Option<Arc<OrchestratorClient>>,
```

Comment:
Is this for the permissioned instance? Because I don't think we can do this for the permissionless...

Reply:
Yes, it is for permissioned instance. A permission node would do this `init_node()` call. It is similar to `init_node` in sequencer but without persistence.